core/src/tile/tileWorker.cpp

```diff
@@ -78,26 +108,17 @@ void TileWorker::run() {
             m_queue.erase(it);
         }
 
-        if (task->isCanceled()) { continue; }
-
-        // Save shared reference to Scene while building tile
-        // FIXME: Scene could be released on Worker-Thread and
-        // therefore call unsafe glDelete* functions...
-        auto scene = m_scene;
-        if (!scene) { continue; }
+        if (task->isCanceled()) {
+            continue;
+        }
 
-        auto tileData = task->source().parse(*task, *scene->mapProjection());
+        auto tileData = task->source().parse(*task, *builder->scene().mapProjection());
 
         // const clock_t begin = clock();
 
-        context.initFunctions(*scene);
-
         if (tileData) {
-            auto tile = std::make_shared<Tile>(task->tileId(),
-                                               *scene->mapProjection(),
-                                               &task->source());
 
-            tile->build(context, *scene, *tileData, task->source());
+            auto&& tile = builder->build(task->tileId(), *tileData, task->source());
```

Comment:
I would guess that this is a minor optimization to prevent the construction and destruction of a `shared_ptr`.